src/Microsoft.Health.Dicom.SqlServer/Features/Query/SqlQueryGenerator.cs

```diff
@@ -239,11 +371,17 @@ public override void Visit(DateRangeValueMatchCondition rangeValueMatchCondition
             var dicomTagSqlEntry = DicomTagSqlEntry.GetDicomTagSqlEntry(rangeValueMatchCondition.DicomTag);
             var tableAlias = GetTableAlias(dicomTagSqlEntry);
             _stringBuilder
-                .Append("AND ")
+                .Append("AND ");
+
+            tableAlias = AppendCustomTagKeyFilter(dicomTagSqlEntry, tableAlias, rangeValueMatchCondition) ?? tableAlias;
```

Comment:
This code is very confusing to me. You already have the right tableAlias from GetTableAlias()

Why would you not just call AppendCustomTagKeyFilter() without expecting any return

Reply:
The custom tag key is appended to the table alias so that we can join against the same custom tag table multiple times!

---
In reply to: [582517527](https://github.com/microsoft/dicom-server/pull/567#discussion_r582517527) [](ancestors = 582517527)

Reply:
Okay can you add that logic to GetTableAlias?